### PR TITLE
Feature/72373   validate tag no

### PR DESCRIPTION
**Pull request by bratli**
Added check to validate tag no as it is written. Icons used to show user if it is valid or not.

### CODE REVIEW

**Comment by Cnordbo** [comment]
Should we delay the API request until after the user stops typing for something like 100ms? 
right now it fires an API request on every keystroke, which results in 5-10 requests being fired and only 1 of them gives the user any value. 

If we implement the loading indicator as suggested elsewhere, the loading symbol can start as soon as the user changes the input, and end when the API gives a response.

**Comment by Cnordbo** [comment]
We should add some tests. 

1) Test that it renders a 'valid' icon when API says its valid
2) Test that it renders a 'invalid' symbol when API says its invalid

File: src/modules/Preservation/views/AddScope/CreateAreaTag/__tests__/CreateAreaTag.test.jsx [hunk 1]
```diff
@@ -69,5 +77,26 @@ describe('<CreateAreaTag />', () => {
...
+            await waitForElement(() => document.querySelector('.valid'));
+        });
+    });
+
+    it('Displays \'Invalid\' icon when suffix contains space', async () => {
+        await act(async () => {
+            render(<CreateAreaTag suffix="1 2" />);
+            await waitForElement(() => document.querySelector('#tagNumberIcon'));
+            await waitForElement(() => document.querySelector('.invalid'));
+        });
```
[the remark this answers is not in the document]
Comment (Cnordbo) [reply src/modules/Preservation/views/AddScope/CreateAreaTag/__tests__/CreateAreaTag.test.jsx:100]: waitForElement probably only needs to be used on the first element.

File: src/modules/Preservation/views/AddScope/CreateAreaTag/__tests__/CreateAreaTag.test.jsx [hunk 2]
```diff
@@ -69,5 +77,26 @@ describe('<CreateAreaTag />', () => {
...
+            render(<CreateAreaTag />);
+            expect(document.querySelector('#tagNumberIcon')).not.toBeInTheDocument();
+        });
+    });
 
+    it('Displays \'Valid\' icon when properties are set', async () => {
+        await act(async () => {
+            render(<CreateAreaTag area={mockAreas[0]} discipline={mockDisciplines[0]} areaType={{ text: 'Normal', value: 'PreArea' }} suffix="12" />);
+            await waitForElement(() => document.querySelector('#tagNumberIcon'));
+            await waitForElement(() => document.querySelector('.valid'));
```
[the remark this answers is not in the document]
Comment (Cnordbo) [reply src/modules/Preservation/views/AddScope/CreateAreaTag/__tests__/CreateAreaTag.test.jsx:91]: waitForElement probably only needs to be used on the first element.

File: src/modules/Preservation/views/AddScope/CreateAreaTag/CreateAreaTag.tsx [hunk 3]
```diff
@@ -154,6 +162,47 @@ const CreateAreaTag = (props: CreateAreaTagProps): JSX.Element => {
...
+            return false;
+        }
+    };
+
+    useEffect(() => {
+        setLoading(true);
+        const checkTagNos = async () => {
+            if (props.suffix && /\s/.test(props.suffix)) {
+                setTagNoValid(false);
+                setFieldsNeededToCheckTagNoSet(true);
```
Comment (Cnordbo) [inline src/modules/Preservation/views/AddScope/CreateAreaTag/CreateAreaTag.tsx:186]: Can this be removed from here? or should this be moved to an inner if of the `else if` statement  under? 

Just feels wrong to say that "all fields" are valid, when its basically only the suffix that is set.

Comment (bratli) [reply src/modules/Preservation/views/AddScope/CreateAreaTag/CreateAreaTag.tsx:186]: I changed the variable name, so it doesnt say "all fields", or am I missing something?

**Pull request closed without merging**